### ISSUE TITLE
Chat - Preview shows | > | > | when adding multiple > in the composer

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#36227]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4-39-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to any chat
3. Type >>>>> in the composer

## Expected Result:
The composer displays | >>> (Android and iOS app behavior)

1. Limit max depth of the quote inside ExpensiMark
2. change implementation of blockquotes in chat to show multi-level quotes
3. Enable inside react-native-live-markdown library support for multilevel quotes on all platforms - here won't be any limit, as parser should handle it.

Discussion for expected result https://expensify.slack.com/archives/C01GTK53T8Q/p1707904106957409

## Actual Result:
The composer shows | > | > | instead of | >>>>>

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/05015ed7-40c3-45bb-acf5-933a40b824ca

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp
CC @quinthar

**Comment by lakchote** [comment]
Offending [PR](https://github.com/Expensify/App/pull/35837) found. This is the one related to markdown preview.

cc @shubham1206agra @tomekzaw @thienlnam

**Comment by tomekzaw** [comment]
Thanks for reporting this, we will decide on the expected behavior here and implement it.

**Comment by thienlnam** [comment]
We've reverted the offending PR https://github.com/Expensify/App/pull/35837, so this is no longer a blocker

**Comment by thienlnam** [comment]
Discussion for expected outcome here: https://expensify.slack.com/archives/C01GTK53T8Q/p1707904106957409
The new plan from @robertKozik:
1. Limit max depth of the quote inside ExpensiMark
2. change implementation of blockquotes in chat to show multi-level quotes
3. Enable inside react-native-live-markdown library support for multilevel quotes on all platforms - here won't be any limit, as parser should handle it.

**Comment by robertKozik** [comment]
Hi! Small update on my end.
I've managed to create draft PR with ExpensiMark changes (Here: https://github.com/Expensify/expensify-common/pull/654) I've come across the undiscovered issue, that parser was removing one excess `>` every time - it slowed me down a little, but I've managed to fix it as well.

Now I'm on adjusting `react-native-live-markdown`, as I think should be merged before, and after them we should update E/App

**Comment by robertKozik** [comment]
The support for multiple quotes has already been merged into the `react-native-live-markdown` library.

The PR within` expensify-common` is ready for review, and once merged, it will be the final piece needed to fully support nested quotes.

**Comment by lakchote** [comment]
Any update @robertKozik? The PR in `expensify-common` has been merged.

**Comment by thienlnam** [comment]
Bump on this - looks like we just need to bump a version or something?

**Comment by robertKozik** [comment]
Yeah, sorry for my absence in this thread. New version of the expensify-common is already merged in the live-markdown library. Version bump for E/App will take place inside the PR with new version of `react-native-live-markdown` for the App

**Comment by bernhardoj** [comment]
Hi, I have a live markdown PR [here](https://github.com/Expensify/react-native-live-markdown/pull/242) to fix this [app issue](https://github.com/Expensify/App/issues/37533) and I notice that the current version of live markdown in app is 0.1.5, but the latest version of live markdown is 0.1.28.

@robertKozik I don't see the App PR yet, so just want to confirm, is your App PR will update the live markdown version to 0.1.28?

(our plan is to wait for the live markdown version to be updated to 0.1.28 and then I will separately create another PR to update it further to include my live markdown PR above. I was thinking to ask for your help to include my live markdown PR above, but I think it would be out of scope)

**Comment by BartoszGrajdek** [comment]
@thienlnam I believe this is now fixed 🤔

**Comment by BartoszGrajdek** [comment]
This seems to not be a problem anymore @thienlnam can we get someone to test it again? 🤔

**Comment by kavimuru** [comment]
@thienlnam Issue is fixed. 

https://github.com/Expensify/App/assets/43996225/bff25944-af74-406c-a43a-2e24fb149f2a

**Comment by thienlnam** [comment]
We're good, thanks!
